src/simulations/initial_conditions.jl

```diff
@@ -42,45 +44,93 @@ function set_soil_initial_conditions!(
     )
 
     Y.soil.ϑ_l .= enforce_residual_constraint.(Y.soil.ϑ_l, θ_r)
-    Y.soil.θ_i .= enforce_porosity_constraint.(Y.soil.ϑ_l, Y.soil.θ_i, ν, θ_r)
+    Y.soil.ϑ_l .= enforce_porosity_constraint.(Y.soil.ϑ_l, ν)
+    Y.soil.θ_i .=
+        enforce_residual_constraint.(Y.soil.θ_i, eltype(Y.soil.θ_i)(0))
+    Y.soil.θ_i .= enforce_porosity_constraint.(Y.soil.ϑ_l, Y.soil.θ_i, ν)
+    ρc_s =
+        ClimaLand.Soil.volumetric_heat_capacity.(
+            Y.soil.ϑ_l,
+            Y.soil.θ_i,
+            soil.parameters.ρc_ds,
+            soil.parameters.earth_param_set,
+        )
     Y.soil.ρe_int .= SpaceVaryingInput(
         soil_ic_path,
         "sie",
         subsurface_space;
         regridder_type,
         regridder_kwargs = (; extrapolation_bc,),
     )
+    T =
+        ClimaLand.Soil.temperature_from_ρe_int.(
+            Y.soil.ρe_int,
+            Y.soil.θ_i,
+            ρc_s,
+            soil.parameters.earth_param_set,
+        )
+    T .= clip_to_atmos_extrema.(T, T_bounds[1], T_bounds[2])
+    Y.soil.ρe_int .= ClimaLand.Soil.volumetric_internal_energy.(
+        Y.soil.θ_i,
+        ρc_s,
+        T,
+        soil.parameters.earth_param_set,
+    )
     return nothing
 end
 
+"""
```

Comment:
Will add doc string

Reply:
(I just dont want to push because I dont want to trigger a long run, but want to keep the latest build with the long run that ran!)